### PR TITLE
test: improve ViewLabs.test

### LINKED ISSUE #245: Improve labs/ViewLabs.test

### PR DESCRIPTION
**Pull request by codyarose**
Fixes #245

### CODE REVIEW

File: src/__tests__/labs/ViewLabs.test.tsx [hunk 1]
```diff
@@ -41,22 +41,21 @@ const setup = (permissions: Permissions[] = []) => {
...
+        <Router history={history}>
+          <ButtonBarProvider.ButtonBarProvider>
+            <ButtonToolbar />
+            <titleUtil.TitleProvider>
+              <ViewLabs />
+            </titleUtil.TitleProvider>
+          </ButtonBarProvider.ButtonBarProvider>
+        </Router>
+      </Provider>,
+    ),
```
Comment (JacobMGEvans) [inline src/__tests__/labs/ViewLabs.test.tsx:58]: 👍🏻

File: src/__tests__/labs/ViewLabs.test.tsx [hunk 2]
```diff
@@ -125,30 +120,36 @@ describe('View Labs', () => {
...
+        id: '5678',
+        type: 'another type',
+        patient: 'patientIdB',
+        status: 'requested',
+        requestedOn: '2020-03-30T04:43:20.102Z',
+      } as Lab
+      const expectedSearchText = 'another'
+      const { expectedLab } = setup([Permissions.ViewLabs, Permissions.RequestLab])
+
+      jest.spyOn(LabRepository, 'findAll').mockResolvedValue([expectedLab2])
```
Comment (JacobMGEvans) [inline src/__tests__/labs/ViewLabs.test.tsx:140]: Is this doable without a spy?

File: src/__tests__/labs/ViewLabs.test.tsx [hunk 3]
```diff
@@ -125,30 +120,36 @@ describe('View Labs', () => {
...
+      expect(await screen.findByRole('cell', { name: expectedLab.code })).toBeInTheDocument()
       jest.useFakeTimers()
-      setup([Permissions.ViewLabs])
 
-      const expectedSearchText = 'search text'
       userEvent.type(screen.getByRole('textbox', { name: /labs.search/i }), expectedSearchText)
 
       act(() => {
         jest.advanceTimersByTime(500)
       })
```
Comment (JacobMGEvans) [inline src/__tests__/labs/ViewLabs.test.tsx]: Is there a way to do this with `waitFor()` the expect or something?

File: src/__tests__/labs/ViewLabs.test.tsx (hunk 2 above)
Comment (codyarose) [reply src/__tests__/labs/ViewLabs.test.tsx:140]: Setting that spy is the only way I've been able to get this to pass with results that resemble how the search is supposed to behave

File: src/__tests__/labs/ViewLabs.test.tsx (hunk 3 above)
Comment (codyarose) [reply src/__tests__/labs/ViewLabs.test.tsx]: Looks like it will pass if I replace `act` with `await waitFor`, is that what you're talking about?

Comment (JacobMGEvans) [reply src/__tests__/labs/ViewLabs.test.tsx]: I was thinking more like
```js
      await wairFor(() => {
        expect(await finBy*(whatever))
      })
```

Comment (nobrayner) [reply src/__tests__/labs/ViewLabs.test.tsx]: An await in an await? :sweat_smile: 

Either:
```javascript
await waitFor(() => {
  expect(screen.getBy*())
}
```
Or
```javascript
expect(await screen.findBy*())
```

Comment (JacobMGEvans) [reply src/__tests__/labs/ViewLabs.test.tsx]: what never seen async/await hell lmao

File: src/__tests__/labs/ViewLabs.test.tsx [hunk 4]
```diff
@@ -85,24 +84,20 @@ describe('View Labs', () => {
...
+      const headers = await screen.findAllByRole('columnheader')
+      const cells = screen.getAllByRole('cell')
+
+      expect(headers[0]).toHaveTextContent(/labs\.lab\.code/i)
+      expect(headers[1]).toHaveTextContent(/labs\.lab\.type/i)
+      expect(headers[2]).toHaveTextContent(/labs\.lab\.requestedOn/i)
+      expect(headers[3]).toHaveTextContent(/labs\.lab\.status/i)
+      expect(cells[0]).toHaveTextContent(expectedLab.code)
+      expect(cells[1]).toHaveTextContent(expectedLab.type)
+      expect(cells[2]).toHaveTextContent(
```
Comment (nobrayner) [inline src/__tests__/labs/ViewLabs.test.tsx:96]: I'm not sure about this... This is now enforcing the order of of the columns - which may or may not be important :man_shrugging:

Comment (JacobMGEvans) [reply src/__tests__/labs/ViewLabs.test.tsx:96]: > I'm not sure about this... This is now enforcing the order of the columns - which may or may not be important

If it's mock data then it is irrelevant, you would still just be asserting things are rendering. 😄

**Pull request merged**